packages/js/src/logger.ts

```diff
@@ -0,0 +1,95 @@
+import { Axiom } from "./client";
+import { LogEvent, LogLevel, LoggerOptions } from "./type";
+import { jsonFriendlyErrorReplacer, prettyPrint, throttle } from "./utils";
+
+class Logger {
+    public dataset: string;
+    private logLevel: LogLevel;
+    private LOG_LEVEL = process.env.AXIOM_LOG_LEVEL || 'debug';
+    public logEvents: LogEvent[] = [];
+    children: Axiom[] = [];
+    throttledSendLogs = throttle(this.sendLogs.bind(this), 1000);
+
+    constructor(public client: Axiom, public options: LoggerOptions) {
+        this.dataset = options.dataset;
+        this.logLevel = this.calculateLogLevel(options.logLevel);
+    }
+
+    debug = (message: string, args: { [key: string]: any } = {}) => {
+        this._log(LogLevel.debug, message, args);
+    };
+
+    info = (message: string, args: { [key: string]: any } = {}) => {
+        this._log(LogLevel.info, message, args);
+    };
+
+    warn = (message: string, args: { [key: string]: any } = {}) => {
+        this._log(LogLevel.warn, message, args);
+    };
+
+    error = (message: string, args: { [key: string]: any } = {}) => {
+        this._log(LogLevel.error, message, args);
+    };
+
+    private _log = (level: LogLevel, message: string, args: { [key: string]: any } = {}) => {
+        if (level < this.logLevel) {
+            return;
+        }
+        const logEvent: LogEvent = {
+            level: LogLevel[level].toString(),
+            message,
+            _time: new Date(Date.now()).toISOString(),
+            fields: this.options.args || {},
+        };
+        if (args instanceof Error) {
+            logEvent.fields = { ...logEvent.fields, message: args.message, stack: args.stack, name: args.name };
+        } else if (typeof args === 'object' && args !== null && Object.keys(args).length > 0) {
+            const parsedArgs = JSON.parse(JSON.stringify(args, jsonFriendlyErrorReplacer));
+            logEvent.fields = { ...logEvent.fields, ...parsedArgs };
+        } else if (args && args.length) {
+            logEvent.fields = { ...logEvent.fields, args: args };
+        }
+        this.logEvents.push(logEvent);
+        if (this.options.autoFlush) {
+            this.throttledSendLogs();
+        }
+    };
+
+    private sendLogs() {
+        console.log("sending logs")
+        if (!this.logEvents.length) {
+            return;
+        }
+        const dataset = this.options.dataset;
+
+        if (!dataset) {
+            this.logEvents.forEach((ev) => prettyPrint(ev));
+            this.logEvents = [];
+            return;
+        }
+        const currentLogs = this.logEvents;
+        this.logEvents = [];
+
+        try {
+            this.client.ingest(dataset, currentLogs);
+        } catch (error) {
+            console.error("Error while ingesting logs:", error);
+        }
+        currentLogs.forEach((log) => prettyPrint(log));
```

Comment:
why are we printing the logs again to the console?

Reply:
I  think printing to the console for debugging purpose it might be useful, what do you think

Reply:
yes, but not everyone wants that on production, I think they have to opt in/out for that.